src/LargeFileUploadTask.ts

```diff
@@ -0,0 +1,338 @@
+    rangeSize?: number;
+    maxTries?: number;
+};
+
+/**
+ * Signature for upload session resulting from the session creation in the server
+ */
+interface LargeFileUploadSession {
+    url: string;
+    expiry: Date;
+}
+
+/**
+ * Signature for the response results in the status enquiry request
+ */
+interface UploadStatusResponse {
+    expirationDateTime: string;
+    nextExpectedRanges: string[];
+}
+
+/**
+ * Holds the details of file required
+ */
+interface FileObject {
+    content: File | ArrayBuffer;
+    name: string;
+    size: number;
+}
+
+/**
+ * Class representing LargeFileUploadTask
+ */
+export class LargeFileUploadTask {
+    /** The GraphClient instance */
+    client: Client
+
+    /** The object holding file details */
+    file: FileObject
+
+    /** The object holding options for the task  */
+    options: LargeFileUploadTaskOptions
+
+    /** The object for upload session */
+    uploadSession: LargeFileUploadSession
+
+    /** The next range needs to be uploaded */
+    nextRange: Range
+
+    /**
+     * Contructs a LargeFileUploadTask
+     * @param client - The GraphClient instance
+     * @param file - The FileObject holding file needs to be uploaded
+     * @param uploadSession - The upload session to which the upload has to be done
+     * @param options - The upload task option
+     */
+    constructor(client: Client, file: FileObject, uploadSession: LargeFileUploadSession, options: LargeFileUploadTaskOptions) {
+        let self = this;
+        self.client = client;
+        self.file = file;
+        options.rangeSize = getValidRangeSize(options.rangeSize);
+        self.options = options;
+        self.uploadSession = uploadSession;
+        self.nextRange = new Range(0, self.options.rangeSize - 1);
+    }
+
+    /**
+     * @static
+     * @async
+     * Creates a LargeFilUploadTask
+     * @param client - The GraphClient instance
+     * @param file - File representated as Blob, File or NodeFile object
+     * @param options - The options for upload task
+     * @return The promise that will be resolves to LargeFileUploadTask instance
+     */
+    static async create(client: Client, file: Blob | File | Buffer, options: LargeFileUploadTaskOptions): Promise<any> {
+        let self = LargeFileUploadTask,
+            _fileObject: FileObject = <FileObject>{};
+        console.log(file.constructor.name);
+        debugger;
+        switch(file.constructor.name) {
+            case "Blob":
+                _fileObject.name = (options.fileName !== undefined) ? options.fileName : self.getRandomFileName();
+                _fileObject.content = new File([<Blob>file], _fileObject.name);
+                _fileObject.size = _fileObject.content.size;
+                break;
+            case "File":
+                let _file = <File>file;
+                _fileObject.content = _file;
+                _fileObject.name = _fileObject.name = (options.fileName !== undefined) ? options.fileName : _file.name;
```

Comment:
Extra '_fileObject.name =' or is this something I don't understand?

Can _file.name ever be undefined? If so, we should use the random name as a last resort.

Reply:
>Can _file.name ever be undefined? If so, we should use the random name as a last resort.

_file.name will never be undefined.